ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLFlowAgentRunner.java

```diff
@@ -168,14 +170,14 @@ private void updateInteraction(Map<String, Object> additionalInfo, String intera
             );
     }
 
-    private String parseResponse(Object output) {
+    private String parseResponse(Object output) throws IOException {
         if (output instanceof List && !((List) output).isEmpty() && ((List) output).get(0) instanceof ModelTensors) {
             ModelTensors tensors = (ModelTensors) ((List) output).get(0);
-            Object response = tensors.getMlModelTensors().get(0).getDataAsMap().get("response");
-            return response + "";
-
+            return tensors.toXContent(JsonXContent.contentBuilder(), null).toString();
         } else if (output instanceof ModelTensor) {
-            return toJson(((ModelTensor) output).getDataAsMap());
+            return ((ModelTensor) output).toXContent(JsonXContent.contentBuilder(), null).toString();
+        } else if (output instanceof ModelTensorOutput) {
+            return ((ModelTensorOutput) output).toXContent(JsonXContent.contentBuilder(), null).toString();
```

Comment:
Looks good to me, we can combine ModelTensor and  ModelTensorOutput together to make it more simpler like

```java
if (output instance of ToXContentObject) {
   return ((ToXContentObject) output).toXContent(JsonXContent.contentBuilder(), null).toString();
}
```